c_src/quicer_config.c

```diff
@@ -768,7 +769,14 @@ getopt3(ErlNifEnv *env,
 
   if (IS_SAME_TERM(ATOM_QUIC_GLOBAL, ctx))
     {
-      res = get_global_opt(env, NULL, eopt);
+      pthread_mutex_lock(&MsQuicLock);
+      // Get global option without using any ctx
+      // We are risking using a closed lib
```

Comment:
In a env that when there is no NIF resource is allocated (reg, conf, listener, conn, stream), VM will unload the module while calling the unload DSO in parallel.